src/mail.sh

```diff
@@ -1056,7 +1057,14 @@ function parse_mail_options()
           commit_count="${BASH_REMATCH[0]}"
           options_values['COMMIT_RANGE']+="$commit_count "
         fi
-        options_values['PASS_OPTION_TO_SEND_EMAIL']="$(str_strip "$* ${options_values['PATCH_VERSION']}")"
+        for i in "$@"; do
+          if [[ "${i}" =~ " " ]]; then
+            pass_option_to_send_email+=" ${i@Q}"
+          else
```

Comment:
1. I could not understand.
2. `"PATCH i-g-t" -3` should be converted to `'PATCH i-g-t'` `'-3'` because the shell will "put" `PATCH i-g-t` to the first argument ($1) and `-3` to the second ($2) (could be other numbers but its besides the point). The expansion of the `$@` is special, it expands to positional arguments ($1, $2, $3....) and not a string with those positional arguments. This way we can be sure the correct quotation of the arguments.

Reply:
Note that the shell strips the outside quotes from `"PATCH i-g-t"` but sends the entire string (with the black space) to be the positional argument.